frontend/pages/payment/[transaction_id].tsx

```diff
@@ -0,0 +1,165 @@
+import { Elements } from '@stripe/react-stripe-js'
+import { loadStripe, Stripe } from '@stripe/stripe-js'
+import { GetStaticPaths, GetStaticProps } from 'next'
+import { useTranslation } from 'next-i18next'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+import { NextSeo } from 'next-seo'
+import { useRouter } from 'next/router'
+import { ReactElement, useEffect, useState } from 'react'
+import Main from '../../src/components/layout/Main'
+import Checkout from '../../src/components/payment/checkout/Checkout'
+import Spinner from '../../src/components/Spinner'
+import { useUserContext } from '../../src/context/user-info'
+import {
+  STRIPE_DATA_URL,
+  TRANSACTION_INFO_URL,
+  TRANSACTION_STRIPE_INFO_URL,
+} from '../../src/env'
+import { TransactionDetailed } from '../../src/types/Payment'
+
+// Memoized Stripe object retrieval so it's only retrieved on demand
+let stripePromise: Promise<Stripe>
+async function getStripeObject() {
+  if (!stripePromise) {
+    let res: Response
+    try {
+      res = await fetch(STRIPE_DATA_URL)
+    } catch {
+      throw 'network-error-try-again'
+    }
+
+    if (res.ok) {
+      const stripeData = await res.json()
+      stripePromise = loadStripe(stripeData.public_key)
+      return stripePromise
+    } else {
+      throw 'network-error-try-again'
+    }
+  } else {
+    return stripePromise
+  }
+}
+
+async function getTransaction(txnId: string) {
```

Comment:
Is `txnId` known in the stripe context? Otherwise I would like to not have abbreviations :)

Reply:
Was following the backend naming, but will expand these